### PR TITLE
Search

### PR DESCRIPTION
**Pull request by weyj4**

### CODE REVIEW

**Comment by teddyknox** [comment]
Thanks for the pull. Can you merge master into your branch, fix the merge conflicts and push back up?

File: app/controllers/search_controller.rb [hunk 1]
```diff
@@ -1,10 +1,10 @@
...
-    if not params[:q].nil?
-      @posts = Post.search(params[:q]).map {|p| Post.find(p['_source']['id'])}
-     else 
-      @posts = []
-    end
-    render json: @posts, include: [:author]
-  end
+	def search
+	  if not params[:q].nil?
+	    posts = Post.search(params[:q]).map{ |p| p["_source"]}
```
Comment (teddyknox) [inline app/controllers/search_controller.rb]: Could you correct the indentation to be consistent here?

File: config.ru [hunk 2]
```diff
@@ -2,3 +2,12 @@
 
 require ::File.expand_path('../config/environment', __FILE__)
 run Rails.application
+
+require 'rack/cors'
+
+    use Rack::Cors do
```
Comment (teddyknox) [inline config.ru]: Correct indentation please.

File: lib/tasks/elastic_search.rake [hunk 3]
```diff
@@ -1,20 +1,10 @@
 namespace :elastic_search do
   desc "Configuring Elastic Search"
   task create: [:environment] do
-    Answer.__elasticsearch__.create_index! force: true
-    Author.__elasticsearch__.create_index! force: true
     Post.__elasticsearch__.create_index! force: true
```
Comment (teddyknox) [inline lib/tasks/elastic_search.rake]: Are we sure that we want the "force" option enabled here?

File: lib/tasks/setup.rake [hunk 4]
```diff
@@ -1,13 +1,25 @@
 namespace :setup do
 
-  task :run, [:environment] do
+  task :migrate, [:environment] do
```
Comment (teddyknox) [inline lib/tasks/setup.rake]: Why are we renaming all of these tasks? Maybe we should just put them all under the same namespace.

File: lib/tasks/elastic_search.rake (hunk 3 above)
Comment (weyj4) [inline lib/tasks/elastic_search.rake]: I'm looking into it

File: lib/tasks/setup.rake [hunk 5]
```diff
@@ -1,11 +1,25 @@
 namespace :setup do
+
   task :run, [:environment] do
```
Comment (teddyknox) [inline lib/tasks/setup.rake:3]: Why are we renaming these tasks?

Comment (weyj4) [inline lib/tasks/setup.rake:3]: Don't know. I didn't mess with the setup file, when I pulled from the origin I had to resolve a merge conflict, that's what I ended up with. Should I change it back to :migrate?

Comment (teddyknox) [inline lib/tasks/setup.rake:3]: I'll update it so that it makes sense.

Teddy

Comment (teddyknox) [inline lib/tasks/setup.rake:3]: I have a feeling that Potter did this, so I'll leave it.

**Pull request merged**